resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/ExecutorPodFactory.scala

```diff
@@ -108,6 +109,8 @@ private[spark] class ExecutorPodFactory(
       nodeToLocalTaskCount: Map[String, Int]): Pod = {
     val name = s"$executorPodNamePrefix-exec-$executorId"
 
+    val imagePullSecrets = imagePullSecret.map(new LocalObjectReference(_)).toList
```

Comment:
Given the same code is used to configure both the driver and executor pods, it can be extracted out into a utility method in `KubernetesUtil`.

Reply:
@liyinan926 I made the change, but imo it looks a bit awkward, as what is done here is just type conversion `Option[String] ~> List[LocalObjectReference]` to make config option palatable for the k8s.

Reply:
Yeah, it seems a bit awkward. I guess the original form looks better. Sorry for that, but looks like it's better if you revert this change.

Reply:
@liyinan926 reverted! Should be good to merge now. Rebased against master too.